### PR TITLE
builtin: add compress funciton

### PR DESCRIPTION
**Pull request by liyu4**
add compress function according to “zlib” algorithm

### CODE REVIEW

File: expression/builtin_encryption.go [hunk 1]
```diff
@@ -293,7 +296,27 @@ type builtinCompressSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/encryption-functions.html#function_compress
 func (b *builtinCompressSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("COMPRESS")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
```
Comment (shenli) [inline expression/builtin_encryption.go]: return d, errors.Trace(err)

File: expression/builtin_encryption_test.go [hunk 2]
```diff
@@ -161,3 +161,28 @@ func (s *testEvaluatorSuite) TestMD5(c *C) {
...
+	fc := funcs[ast.Compress]
+	for _, test := range compressCases {
+		arg := types.NewDatum(test.in)
+		f, err := fc.getFunction(datumsToConstants([]types.Datum{arg}), s.ctx)
+		c.Assert(err, IsNil)
+		out, err := f.eval(nil)
+		c.Assert(err, IsNil)
+		c.Assert(out, DeepEquals, types.NewDatum(test.out))
+	}
+	s.testNullInput(c, ast.AesDecrypt)
```
Comment (shenli) [inline expression/builtin_encryption_test.go]: ast.Compress

File: plan/typeinferer.go [hunk 3]
```diff
@@ -406,6 +406,9 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
 		tp = types.NewFieldType(mysql.TypeVarString)
 		chs = v.defaultCharset
 		tp.Flen = 32
+	case ast.Compress:
+		tp = types.NewFieldType(mysql.TypeBlob)
+		chs = v.defaultCharset
```
Comment (shenli) [inline plan/typeinferer.go]: Blob should use binary as charset.

**Comment by liyu4** [comment]
@shenli  thank you, I had modified some code according to your suggest

File: expression/builtin_encryption.go [hunk 4]
```diff
@@ -293,7 +296,27 @@ type builtinCompressSig struct {
...
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	arg := args[0]
+	if arg.IsNull() {
+		return d, nil
+	}
+
+	compressStr, err := args[0].ToBytes()
```
Comment (ngaut) [inline expression/builtin_encryption.go]: s/args[0]/arg/

Comment (liyu4) [inline expression/builtin_encryption.go]: got it

**Comment by liyu4** [comment]
@ngaut thanks

**Comment by ngaut** [comment]
LGTM, thank you.

**Pull request merged**